test/unit/api.js

```diff
@@ -85,41 +88,122 @@ define([
         equal(originalEvent.type, 'original', 'original event.type is not modified');
     });
 
-    test('has chainable methods', function(assert) {
+    test('defines methods', function(assert) {
         var api = createApi('player');
 
-        var chainable = {
-            // COMMENTED OUT UNTIL FIXED:
-            setup: [{}],
-            //load: [{}],
-            //forceState: [''],
-            //play: void 0,
-            //seek: [0],
-            //setControls: [true],
-            //setFullscreen: [false],
-            //playlistNext: void 0,
-            //playlistPrev: void 0,
-            //playlistItem: [0],
-            //setCurrentCaptions: [{}],
-            //setCurrentQuality: [0],
-            //setCurrentAudioTrack: [0],
-            //setVolume: [100],
-            //setMute: [false],
-            //resize: [100, 75],
-            on: [''],
-            once: [''],
-            trigger: [''],
-            off: void 0,
-            remove: void 0
-        };
-
-        _.each(chainable, function(args, method) {
+        _.each(apiMethods, function(args, method) {
             assert.ok(_.isFunction(api[method]), 'api.' + method + ' is defined');
+        });
+    });
+
+    test('has chainable methods', function(assert) {
+        var api = createApi('player');
+
+        _.each(apiMethodsChainable, function(args, method) {
             var result = api[method].apply(api, args);
             assert.strictEqual(result, api, 'api.' + method + ' returns an instance of itself');
         });
     });
 
+    test('has getters that return values before setup', function(assert) {
+        var api = createApi('player');
+
+        assert.strictEqual(api.getContainer(), document.getElementById('player'),
+            'getContainer returns the player DOM element before setup');
+
+
+        var result = api.registerPlugin('', '7.0', function(){});
+        assert.strictEqual(result, undefined, 'registerPlugin returns undefined');
+
+        assert.deepEqual( api.getMeta(), {}, 'getMeta returns {}');
+        assert.strictEqual( api.getItem(), undefined, 'getItem returns undefined');
+        assert.strictEqual( api.getPlaylist(), undefined, 'getPlaylist returns undefined');
+        assert.strictEqual( api.getPlaylistItem(), undefined, 'getPlaylistItem() returns undefined');
+        assert.strictEqual( api.getPlaylistItem(0), null, 'getPlaylistItem(0) returns null');
+
+        // FIXME: These are not ready until after setup (controller.setup())
+        //api.qoe();
+        //api.createInstream();
+        //assert.strictEqual( api.getState(), undefined, 'getState returns undefined before setup');
+
+    });
+
+    test('sets up a player with setup(config)', function(assert) {
+        var done = assert.async();
```

Comment:
We talked about this the goals are to
1. Get quicker feedback for problems than cucumber provides
2. Be able to visualize code paths from the API using karmas code coverage tools